diff --git a/.git/hooks/prepare-commit-msg b/.git/hooks/prepare-commit-msg
new file mode 100755
index 0000000..51ea980
--- /dev/null
+++ b/.git/hooks/prepare-commit-msg
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+old_commit_msg="$(<"$1")"
+
+if [ -n "$(sed -n -e '1 s/[[:space:]]*//p' "$1")" ]; then
+	exit
+fi
+
+echo -n "" > "$1"
+
+# added
+for path in $(git diff --name-only --cached --diff-filter=A); do
+	echo "${path%/*}:" >> "$1"
+done
+
+# modified
+for path in $(git diff --name-only --cached --diff-filter=M); do
+	echo "${path%/*}:" >> "$1"
+done
+
+# deleted
+for path in $(git diff --name-only --cached --diff-filter=D); do
+	echo "${path%/*}:" >> "$1"
+done
+
+echo "$old_commit_msg" >> "$1"
